core/src/main/java/io/aiven/klaw/validation/PermissionAllowedValidatorImpl.java

```diff
@@ -0,0 +1,41 @@
+package io.aiven.klaw.validation;
+
+import static io.aiven.klaw.helpers.UtilMethods.getPrincipal;
+
+import io.aiven.klaw.error.PermissionConstraintException;
+import io.aiven.klaw.model.enums.PermissionType;
+import io.aiven.klaw.service.CommonUtilsService;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+@Slf4j
+public class PermissionAllowedValidatorImpl
+    implements ConstraintValidator<PermissionAllowed, Object> {
+  private PermissionType[] permissions;
+  @Autowired private CommonUtilsService commonUtilsService;
+
+  @Value("${klaw.api.permissions.guard.enabled:true}")
+  private boolean checkEnabled;
+
+  @Override
+  public void initialize(PermissionAllowed constraintAnnotation) {
+    this.permissions = constraintAnnotation.permissionAllowed();
+  }
+
+  @Override
+  public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
+    log.debug("Checking for permission(s) {}", permissions);
```

Comment:
permissions argument is not known value

Reply:
it should be set in the Initialize method but i will take a look again in the morning